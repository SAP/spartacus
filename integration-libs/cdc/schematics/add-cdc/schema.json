{
  "$schema": "http://json-schema.org/schema",
  "$id": "CDCSchematics",
  "title": "CDC Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the CDC feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["CDC", "CDC-B2B"],
        "type": "string"
      },
      "x-prompt": {
        "message": "Which feature would you like to set up from the CDC library?",
        "type": "list",
        "items": [
          {
            "value": "CDC",
            "label": "CDC for B2C"
          },
          {
            "value": "CDC-B2B",
            "label": "CDC for B2B"
          }
        ]
      }
    },
    "javascriptUrl": {
      "type": "string",
      "description": "Location of the CDC Java Script SDK in Customer Data Cloud",
      "x-prompt": "[CDC] What is the CDC JS SDK URL?"
    },
    "baseSite": {
      "type": "string",
      "description": "Name of the Spartcus base site",
      "x-prompt": "[CDC] What is the base site?",
      "default": "electronics-spa"
    },
    "sessionExpiration": {
      "type": "number",
      "description": "Session Expiration time in seconds",
      "x-prompt": "[CDC] What is the CDC Session expiration time (in seconds)?",
      "default": 3600
    },
    "httpHeaderName": {
      "type": "string",
      "description": "Name of HTTP Header key for Reset Password",
      "x-prompt": "[CDC] What is the HTTP Header key for Reset Password?",
      "default": "apikey"
    }
  },
  "required": []
}
