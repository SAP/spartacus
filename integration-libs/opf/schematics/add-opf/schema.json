{
  "$schema": "http://json-schema.org/schema",
  "$id": "OpfSchematics",
  "title": "Opf Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": true
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the opf features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["OPF-Checkout", "OPF-Base"],
        "type": "string"
      },
      "default": ["OPF-Checkout", "OPF-Base"]
    },
    "baseUrl": {
      "type": "string",
      "description": "The base url of Cloud Commerce Adapter for Open Payment Framework integration",
      "x-prompt": "[OPF] What is the base URL (origin) of your OPF Cloud Commerce Adapter?"
    },
    "commerceCloudPublicKey": {
      "type": "string",
      "description": "Commerce Clould public key required for authentication between OPF Cloud Commerce Adapter and Spartacus"
    }
  },
  "required": []
}
