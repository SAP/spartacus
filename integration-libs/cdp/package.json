{
  "name": "@spartacus/cdp",
  "version": "2211.31.0",
  "description": "Customer Data Platform Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "customer data cloud",
    "gigya",
    "cdp"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cdp",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@spartacus/customer-ticketing": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
