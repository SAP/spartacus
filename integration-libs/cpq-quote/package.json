{
  "name": "@spartacus/cpq-quote",
  "version": "2211.31.0",
  "description": "cpq-quote library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "cpq-quote"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cpq-quote",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@spartacus/cart": "2211.31.0",
    "@spartacus/core": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "@spartacus/storefront": "2211.31.0",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
