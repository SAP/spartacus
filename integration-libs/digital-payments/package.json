{
  "name": "@spartacus/digital-payments",
  "version": "2211.31.0",
  "description": "Digital Payments Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "digital payments",
    "payments"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/digital-payments",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/forms": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@ng-select/ng-select": "^12.0.4",
    "@spartacus/cart": "2211.31.0",
    "@spartacus/checkout": "2211.31.0",
    "@spartacus/core": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "@spartacus/storefront": "2211.31.0",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
