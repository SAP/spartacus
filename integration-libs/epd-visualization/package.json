{
  "name": "@spartacus/epd-visualization",
  "version": "2211.31.0",
  "description": "SAP Enterprise Product Development Visualization integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "epd-visualization",
    "2d",
    "3d",
    "visual",
    "spare",
    "part",
    "picking",
    "epd"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/epd-visualization",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/forms": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@sapui5/ts-types-esm": "1.120.1",
    "@spartacus/cart": "2211.31.0",
    "@spartacus/core": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "@spartacus/storefront": "2211.31.0",
    "@spartacus/styles": "2211.31.0",
    "bootstrap": "^4.6.2",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
