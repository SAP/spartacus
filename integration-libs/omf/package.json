{
  "name": "@spartacus/omf",
  "version": "2211.31.0",
  "description": "SAP Order Management Foundation Integration",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "omf"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@ngrx/store": "^17.0.1",
    "@spartacus/core": "2211.31.0",
    "@spartacus/order": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
