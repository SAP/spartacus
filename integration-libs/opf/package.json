{
  "name": "@spartacus/opf",
  "version": "2211.25.1",
  "description": "SAP Open Payment Framework integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "opf"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/opf",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/forms": "^17.0.5",
    "@angular/platform-browser": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@ng-select/ng-select": "^12.0.4",
    "@ngrx/store": "^17.0.1",
    "@spartacus/cart": "2211.25.1",
    "@spartacus/checkout": "2211.25.1",
    "@spartacus/core": "2211.25.1",
    "@spartacus/order": "2211.25.1",
    "@spartacus/pickup-in-store": "2211.25.1",
    "@spartacus/schematics": "2211.25.1",
    "@spartacus/storefront": "2211.25.1",
    "@spartacus/styles": "2211.25.1",
    "@spartacus/user": "2211.25.1",
    "bootstrap": "^4.6.2",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
