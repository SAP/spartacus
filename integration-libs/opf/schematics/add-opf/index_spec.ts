/// <reference types="jest" />

import { RunSchematicTaskOptions } from '@angular-devkit/schematics/tasks/run-schematic/options';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  OPF_BASE_FEATURE_NAME,
  OPF_CHECKOUT_FEATURE_NAME,
  SPARTACUS_CHECKOUT,
  SPARTACUS_OPF,
  SPARTACUS_SCHEMATICS,
  LibraryOptions as SpartacusOpfOptions,
  SpartacusOptions,
  checkoutWrapperModulePath,
  opfFeatureModulePath,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';
const collectionPath = path.join(__dirname, '../collection.json');
const scssFilePath = 'src/styles/spartacus/opf.scss';

describe('Spartacus SAP OPF integration schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_OPF,
    collectionPath
  );

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
    standalone: false,
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: SpartacusOpfOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const checkoutFeatureOptions: SpartacusOpfOptions = {
    ...libraryNoFeaturesOptions,
    features: [OPF_CHECKOUT_FEATURE_NAME],
  };

  const opfFeatureOptions: SpartacusOpfOptions = {
    ...libraryNoFeaturesOptions,
    features: [OPF_BASE_FEATURE_NAME],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      path.join(
        __dirname,
        '../../../../projects/schematics/src/collection.json'
      )
    );

    schematicRunner.registerCollection(
      SPARTACUS_CHECKOUT,
      path.join(
        __dirname,
        '../../../../feature-libs/checkout/schematics/collection.json'
      )
    );

    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'workspace',
      workspaceOptions
    );

    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'application',
      appOptions,
      appTree
    );

    appTree = await schematicRunner.runExternalSchematic(
      SPARTACUS_SCHEMATICS,
      'ng-add',
      { ...spartacusDefaultOptions, name: 'schematics-test' },
      appTree
    );
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner.runSchematic(
        'ng-add',
        { ...libraryNoFeaturesOptions, features: [] },
        appTree
      );
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(opfFeatureModulePath)).toBeFalsy();
    });
  });

  describe('SAP OPF feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          { ...checkoutFeatureOptions, ...opfFeatureOptions },
          appTree
        );
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(opfFeatureModulePath);
        expect(module).toMatchSnapshot();
      });

      describe('styling', () => {
        it('should create a proper scss file', () => {
          const scssContent = appTree.readContent(scssFilePath);
          expect(scssContent).toMatchSnapshot();
        });

        it('should update angular.json', async () => {
          const content = appTree.readContent('/angular.json');
          expect(content).toMatchSnapshot();
        });
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });

      it('should run the proper installation tasks', async () => {
        const tasks = schematicRunner.tasks.filter(
          (task) =>
            task.name === 'run-schematic' &&
            (task.options as RunSchematicTaskOptions<{}>).collection ===
              '@sap/opf'
        );

        expect(tasks.length).toEqual(0);
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(opfFeatureModulePath);
        expect(module).toMatchSnapshot();

        const wrapperModule = appTree.readContent(checkoutWrapperModulePath);
        expect(wrapperModule).toMatchSnapshot();
      });
    });
  });
});

it('should pass schematics tests', async () => {
  expect(true).toEqual(true);
});
