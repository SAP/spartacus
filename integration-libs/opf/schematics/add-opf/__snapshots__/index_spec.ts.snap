// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus SAP OPF integration schematics: ng-add SAP OPF feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, provideConfig } from "@spartacus/core";
import { OPF_QUICK_BUY_FEATURE, OpfQuickBuyRootModule } from "@spartacus/opf/quick-buy/root";

@NgModule({
  declarations: [],
  imports: [
    OpfQuickBuyRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [OPF_QUICK_BUY_FEATURE]: {
        module: () =>
          import('@spartacus/opf/quick-buy').then((m) => m.OpfQuickBuyModule),
      },
    }
  })]
})
export class OpfFeatureModule { }
"
`;

exports[`Spartacus SAP OPF integration schematics: ng-add SAP OPF feature general setup should add the feature using the lazy loading syntax 2`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, provideConfig } from "@spartacus/core";
import { OPF_QUICK_BUY_FEATURE, OpfQuickBuyRootModule } from "@spartacus/opf/quick-buy/root";

@NgModule({
  declarations: [],
  imports: [
    OpfQuickBuyRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [OPF_QUICK_BUY_FEATURE]: {
        module: () =>
          import('@spartacus/opf/quick-buy').then((m) => m.OpfQuickBuyModule),
      },
    }
  })]
})
export class OpfFeatureModule { }
"
`;

exports[`Spartacus SAP OPF integration schematics: ng-add SAP OPF feature general setup should add the feature using the lazy loading syntax 3`] = `""`;

exports[`Spartacus SAP OPF integration schematics: ng-add SAP OPF feature general setup styling should create a proper scss file 1`] = `
"@import "../../styles-config";
@import "@spartacus/opf";
"
`;

exports[`Spartacus SAP OPF integration schematics: ng-add SAP OPF feature general setup styling should update angular.json 1`] = `
"{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "",
  "projects": {
    "schematics-test": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss",
          "standalone": false
        },
        "@schematics/angular:directive": {
          "standalone": false
        },
        "@schematics/angular:pipe": {
          "standalone": false
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:application",
          "options": {
            "outputPath": "dist/schematics-test",
            "index": "src/index.html",
            "browser": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.scss",
              "src/styles/spartacus/opf.scss"
            ],
            "scripts": [],
            "stylePreprocessorOptions": {
              "includePaths": [
                "node_modules/"
              ]
            }
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "3.5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "optimization": false,
              "extractLicenses": false,
              "sourceMap": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "schematics-test:build:production"
            },
            "development": {
              "buildTarget": "schematics-test:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "schematics-test:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.scss",
              "src/styles/spartacus/opf.scss"
            ],
            "scripts": [],
            "stylePreprocessorOptions": {
              "includePaths": [
                "node_modules/"
              ]
            }
          }
        }
      }
    }
  }
}"
`;
